libwebauthn/src/transport/hid/device.rs

```diff
@@ -1,34 +1,25 @@
+use super::channel::HidChannel;
+use super::Hid;
 use async_trait::async_trait;
 use hidapi::DeviceInfo;
 use hidapi::HidApi;
 use std::fmt;
 #[allow(unused_imports)]
 use tracing::{debug, info, instrument};
 
-#[cfg(feature = "virtual-hid-device")]
-use solo::SoloVirtualKey;
-
-use super::channel::HidChannel;
-use super::Hid;
-
 use crate::transport::error::TransportError;
 use crate::transport::Device;
 use crate::webauthn::error::Error;
 
-#[derive(Debug)]
-// SoloVirtualKey is not clone-able, but in test-mode we don't care
-#[cfg_attr(not(feature = "virtual-hid-device"), derive(Clone))]
+#[derive(Debug, Clone)]
 pub struct HidDevice {
     pub backend: HidBackendDevice,
 }
 
-#[derive(Debug)]
-// SoloVirtualKey is not clone-able, but in test-mode we don't care
-#[cfg_attr(not(feature = "virtual-hid-device"), derive(Clone))]
+#[derive(Debug, Clone)]
 pub enum HidBackendDevice {
     HidApiDevice(DeviceInfo),
-    #[cfg(feature = "virtual-hid-device")]
```

Comment:
Would it be easier to feature-gate the code in mod.rs (`pub use`), and here?

It would remove the need for all the gating within `virt`, and the mock device which doesn't provide much value.

Reply:
Then we also have to feature-gate every `match` statement for `HidBackendDevice`, which is why I removed the gate here and opted for the mock implementation instead.
Felt cleaner (aka less places needed, where we need the feature-gate) and somewhat less annoying to work with for me, but I have no strong opinions here. Both options would be fine with me.